contracts/interchain-token-service/src/event.rs

```diff
@@ -13,6 +13,26 @@ pub struct TrustedChainRemovedEvent {
     pub chain: String,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct InterchainTokenDeployedEvent {
+    pub token_id: BytesN<32>,
+    pub token_address: Address,
+    pub minter: Option<Address>,
+    pub name: String,
+    pub symbol: String,
+    pub decimals: u32,
```

Comment:
```suggestion
    pub decimals: u8,
```
since this is not a contracttype?

Reply:
Ah, I see